tests/checks/preflight.go

```diff
@@ -0,0 +1,139 @@
+	// nodesWithCPUManager contains the number of nodes with the CPU manager preset
+	MinimumNodesWithCPUManager *int `json:"minimumNodesWithCPUManager,omitempty"`
+	// isKind tells if the tests are running on kind
+	IsKind *bool `json:"isKind,omitempty"`
+}
+
+func IntrospectCluster() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	util.PanicOnError(err)
+	nodes := util.GetAllSchedulableNodes(virtClient).Items
+	nodeCount := len(nodes)
+	DiscoveredClusterProfile.MinimumSchedulableNodes = &nodeCount
+	if IsOpenShift() {
+		majorVersion := cluster.GetOpenShiftMajorVersion(virtClient)
+		DiscoveredClusterProfile.OpenShiftMajorVersion = &majorVersion
+	}
+	kubernetesVersion, err := cluster.GetKubernetesVersion(virtClient)
+	if err != nil {
+		util.PanicOnError(err)
+	}
+	DiscoveredClusterProfile.KubernetesVersion = &kubernetesVersion
+
+	dualStack, err := libnet.IsClusterDualStack(virtClient)
+	if err != nil {
+		util.PanicOnError(err)
+	}
+	DiscoveredClusterProfile.DualNetworkStack = &dualStack
+
+	withCPUManager := 0
+	for _, node := range nodes {
+		if IsCPUManagerPresent(&node) {
+			withCPUManager++
+		}
+	}
+	DiscoveredClusterProfile.MinimumNodesWithCPUManager = &withCPUManager
+
+	isKind := IsRunningOnKindInfra() || IsRunningOnKindInfraIPv6()
+	DiscoveredClusterProfile.IsKind = &isKind
+
+	encoder := json.NewEncoder(os.Stdout)
+	encoder.SetIndent("", "  ")
+	err = encoder.Encode(&DiscoveredClusterProfile)
+	util.PanicOnError(err)
+}
+
+type DiscoveryError struct {
+	errors []error
+}
+
+func (d *DiscoveryError) Error() string {
+	return fmt.Sprintf("%v", d.errors)
+}
+
+func VerifyClusterExpectations() error {
+	if len(flags.ClusterProfilePath) == 0 {
+		return nil
+	}
+
+	expected := &ClusterProfile{}
+	reader, err := os.Open(flags.ClusterProfilePath)
+	if err != nil {
+		log.DefaultLogger().Reason(err).Critical("Could not find the provided cluster profile file.")
+	}
+	if err := yaml.NewYAMLOrJSONDecoder(reader, 1024).Decode(expected); err != nil {
+		log.DefaultLogger().Reason(err).Critical("Could not decode the provided cluster profile file.")
+	}
+
+	errors := &DiscoveryError{}
+	if expected.OpenShiftMajorVersion != nil {
+		if !reflect.DeepEqual(*expected.OpenShiftMajorVersion, *DiscoveredClusterProfile.OpenShiftMajorVersion) {
+			errors.errors = append(errors.errors, fmt.Errorf("OpenShiftMajorVersion was answered with %v, expected %v", *DiscoveredClusterProfile.OpenShiftMajorVersion, *expected.OpenShiftMajorVersion))
+		}
+	}
+	if expected.KubernetesVersion != nil {
+		if !reflect.DeepEqual(*expected.KubernetesVersion, *DiscoveredClusterProfile.KubernetesVersion) {
+			errors.errors = append(errors.errors, fmt.Errorf("KubernetesVersion was answered with %v, expected %v", *DiscoveredClusterProfile.KubernetesVersion, *expected.KubernetesVersion))
+		}
+	}
+	if expected.DualNetworkStack != nil {
+		if !reflect.DeepEqual(*expected.DualNetworkStack, *DiscoveredClusterProfile.DualNetworkStack) {
```

Comment:
Why `DeepEqual` is needed to compare booleans? Same for the other compares of simple types.

Reply:
Just copy-pase. Will change it.